src/Board.java

```diff
@@ -11,6 +11,40 @@ public class Board {
     private Color currentPlayer = Color.WHITE;
     private ArrayDeque<Move> moveStack= new ArrayDeque<Move>();
 
+    public Board(String gameState, String turn) {
+        this.currentPlayer = turn.equals("white") ? Color.WHITE : Color.BLACK;
+        String[] pieceArr = gameState.split(" ", -1);
+        for (int i = 0; i < pieceArr.length; i++) {
+            if (pieceArr[i].equals("PAWN,w")) {
```

Comment:
these checks could be done with significantly less repeated code. if your input is formatted `PIECE_KIND,color`, you should be able to:

1. Split the input by comma
2. Parse the first part into a `PieceKind`
3. Parse the second part into a color, either black or white
4. And then combine the two into the piece to insert into the board

Then you'll avoid all these really annoying and hard-to-read if statements with a couple nice-looking function calls